backend/app/controllers/surveyors_controller.rb

```diff
@@ -6,7 +6,7 @@ class SurveyorsController < ApplicationController
 
   # GET /surveyors or /surveyors.json
   def index
-    @surveyors = Surveyor.where(params.slice(:city, :zipcode, :state, :role, :status))
+    @surveyors = Surveyor.where(search_params.slice(:city, :zipcode, :state, :role, :status))
```

Comment:
I don't think you need to use `.slice` here since we're already using `permit` to allow these.
```suggestion
    @surveyors = Surveyor.where(search_params)
```